*: use the built-in max/min to simplify the code and fix a bug

studystill opened the pull request:
### Problem

In Go 1.21, the standard library includes built-in [max/min](https://pkg.go.dev/builtin@go1.21.0#max) function, which can greatly simplify the code.

### Solution

...

roman-khimov commented on pkg/core/statesync/mptpool.go:
```diff
@@ -54,10 +54,7 @@ func (mp *Pool) GetBatch(limit int) []util.Uint256 {
-		count = limit
-	}
+	count := min(len(mp.hashes), limit)
 	result := make([]util.Uint256, 0, limit)
```
Not related, but should be `make([]util.Uint256, 0, count)` then.

studystill replied:
Fixed. Thanks.

AnnaShaleva reviewed (commented):
Adjust the commit message to follow https://github.com/nspcc-dev/.github/blob/master/git.md#commit-messages:
```
*: use the built-in max/min to simplify the code and fix a bug 
```

The logic itself LGTM.

AnnaShaleva commented on pkg/core/statesync/mptpool.go:
```diff
@@ -54,11 +54,8 @@ func (mp *Pool) GetBatch(limit int) []util.Uint256 {
-	}
-	result := make([]util.Uint256, 0, limit)
+	count := min(len(mp.hashes), limit)
+	result := make([]util.Uint256, 0, count)
```
And BTW, it's a separate logical change (bugfix) that deserves a separate commit.

studystill replied:
To avoid conflicts, I will wait for this PR to be merged before submitting a new PR to fix the bug with result := make([]util.Uint256, 0, limit).

studystill commented:
> Adjust the commit message to follow https://github.com/nspcc-dev/.github/blob/master/git.md#commit-messages:
> 
> ```
> *: use the built-in max/min to simplify the code and fix a bug 
> ```
> 
> The logic itself LGTM.

Thank you for your guidance. I have reverted the changes here and updated the corresponding commit message.

AnnaShaleva reviewed (approved):
> I have reverted the changes here

Splitting bugfix and max/min changes into two separate commits would be sufficient, but OK.
